src/main/fc/config.c

```diff
@@ -266,6 +266,22 @@ static void validateAndFixConfig(void)
     }
 #endif
 
+    if (rxConfig()->rcInterpolation == RC_SMOOTHING_OFF || rxConfig()->rcInterpolationChannels == INTERPOLATION_CHANNELS_T) {
```

Comment:
This needs to be more complex to include checking the smoothing type added in #6017 

The way the logic works is that it `rc_smoothing_type = INTERPOLATION` then everything behaves as before and all the `rc_interp`* settings apply.  If it `rc_smoothing_type = FILTER` then the `rc_interp` setting (OFF, AUTO, MANUAL) isn't used and the smoothing is always on (but still respects the channel selection).  It's made more complicated because of conditional compilation.  I tested with the changes below and it works as expected:

```
#ifdef USE_RC_SMOOTHING_FILTER
    if ((rxConfig()->rc_smoothing_type == RC_SMOOTHING_TYPE_INTERPOLATION && rxConfig()->rcInterpolation == RC_SMOOTHING_OFF)
#else
    if (rxConfig()->rcInterpolation == RC_SMOOTHING_OFF
#endif
       || rxConfig()->rcInterpolationChannels == INTERPOLATION_CHANNELS_T) {
```